tests/interfaceType.spec.ts

```diff
@@ -196,3 +196,49 @@ describe('interfaceType', () => {
     spy.mockRestore()
   })
 })
+
+it('extensions are inherited and deeply merged by field modifications', () => {
+  const schema = makeSchema({
+    types: [
+      interfaceType({
+        name: 'SomeInterface',
+        resolveType() {
+          return 'B'
+        },
+        definition(t) {
+          t.string('foo', {
+            extensions: {
+              deeply: {
+                foo1: true,
+              },
+            },
+          })
+        },
+      }),
+      objectType({
+        name: 'A',
+        definition(t) {
+          t.implements('SomeInterface')
+          t.modify('foo', {
+            extensions: {
+              deeply: {
+                foo2: true,
+              },
+            },
+          })
+        },
+      }),
+    ],
+    outputs: false,
+    shouldGenerateArtifacts: false,
+  })
+  const A = schema.getType('A') as GraphQLObjectType
+  expect(A.getFields().foo.extensions).toMatchInlineSnapshot(`
+    Object {
+      "deeply": Object {
+        "foo1": true,
+        "foo2": true,
```

Comment:
@tgriesser do you agree extension modifications should deeply merge?

Reply:
I think they should probably stay independent until we have a use case for them to be merged.